testlapack/general.go

```diff
@@ -138,6 +141,41 @@ func randomSchurCanonical(n, stride int, rnd *rand.Rand) blas64.General {
 	return t
 }
 
+// blockedUpperTriGeneral returns a normal random, general matrix in the form
+//
+//            c-k-l  k    l
+//  A =    k [  0   A12  A13 ] if r-k-l >= 0;
+//         l [  0    0   A23 ]
+//     r-k-l [  0    0    0  ]
+//
+//          c-k-l  k    l
+//  A =  k [  0   A12  A13 ] if r-k-l < 0;
+//     r-k [  0    0   A23 ]
+//
+// where the k×k matrix A12 and l×l matrix B13 are non-singular
+// upper triangular. A23 is l×l upper triangular if m-k-l >= 0,
+// otherwise A23 is (m-k)×l upper trapezoidal.
+func blockedUpperTriGeneral(r, c, k, l, stride int, kblock bool, rnd *rand.Rand) blas64.General {
+	t := l
+	if kblock {
+		t += k
+	}
+	ans := nanGeneral(r, c, stride)
```

Comment:
Use `zeros`, otherwise unset elements will be NaN, `tola` and `tolb` will be NaN, and the test will not converge. With `zeros` it will.